spikeinterface/extractors/stream_ibl.py

```diff
@@ -0,0 +1,239 @@
+
+import numpy as np
+import probeinterface as pi
+
+from spikeinterface.core import BaseRecording, BaseRecordingSegment
+from spikeinterface.core.core_tools import define_function_from_class
+
+try:
+    import brainbox
+    from one.api import ONE
+
+    HAVE_BRAINBOX_ONE = True
+except ModuleNotFoundError:
+    HAVE_BRAINBOX_ONE = False
+
+
+class StreamingIblExtractor(BaseRecording):
+    """
+    Stream IBL data as an extractor object.
+
+    Parameters
+    ----------
+    session : str
+        The session ID to extract recordings for.
+        In ONE, this is sometimes referred to as the 'eid'.
+        When doing a session lookup such as
+
+        >>> from one.api import ONE
+        >>> one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+        >>> sessions = one.alyx.rest('sessions', 'list', tag='2022_Q2_IBL_et_al_RepeatedSite')
+
+        each returned value in `sessions` refers to it as the 'id'.
+    stream_name : str
+        The name of the stream to load for the session.
+        These can be retrieved from calling `StreamingIblExtractor.get_stream_names(session="<your session ID>")`.
+    load_sync_channels : bool, default: false
+        Load or not the last channel (sync).
+        If not then the probe is loaded.
+    cache_folder : str, optional
+        The location to temporarily store chunks of data during streaming.
+        The default uses the folder designated by ONE.alyx._par.CACHE_DIR / "cache".
+    remove_cached : bool, default: true
+        Whether or not to remove data from the cache after it is read.
+        If you expect to reuse fetched data many times, and have the disk space available, it is recommended to set this to False.
+
+    Returns
+    -------
+    recording : StreamingIblExtractor
+        The recording extractor which allows access to the traces.
+    """
+
+    extractor_name = "StreamingIbl"
+    has_default_locations = True
+    installed = HAVE_BRAINBOX_ONE
+    mode = "folder"
+    installation_mesg = (
+        "To use the StreamingIblExtractor, install ONE-api and ibllib: \n\n pip install ONE-api\npip install ibllib\n"
+    )
+    name = "stream_ibl"
+
+    @classmethod
+    def get_stream_names(cls, session: str, cache_folder: Optional[Union[Path, str]] = None) -> List[str]:
+        """
+        Convenient retrieval of available stream names.
+
+        Parameters
+        ----------
+        session : str
+            The session ID to extract recordings for.
+            In ONE, this is sometimes referred to as the 'eid'.
+            When doing a session lookup such as
+
+            >>> from one.api import ONE
+            >>> one = ONE(base_url="https://openalyx.internationalbrainlab.org", password="international", silent=True)
+            >>> sessions = one.alyx.rest('sessions', 'list', tag='2022_Q2_IBL_et_al_RepeatedSite')
+
+            each returned value in `sessions` refers to it as the 'id'.
+
```

Comment:
Here I would add some documentaion on cache folder.
On my machine the cache folder machinery of ONE push everythin in my home linux.
`/home/samuel/Downloads/ONE/openalyx.internationalbrainlab.org/cache`

In my institute home are very very small!! I woudl easily saturate the home by mistake.
Maybe we should have clear doc on this.

Do you know what is cache eactly here ?
Only metadata or on all download data chunks ?

Reply:
I enhanced the docstring a bit to clarify some of these things.

The cache folder is apparently necessary, all metadata about session IDs, probe names, the AP/LF band meta header files all go here on-demand.

It does indeed seem to tend to choose the 'Downloads' folder on a given system, which you're right, may be quite small.

This is also why I have `remove_cached=True` set as the default, whereas they keep it as `False` by default. That's probably best for performance when repeatedly reusing the same snippets of data multiple times, but when loading a full sessions worth (> 100 GB) that could very easily fill a home volume.

With `remove_cached=True`, each 'chunk' of data (~15 MB x the number of chunks you need to stream to return the specified slice of the traces) gets removed after you read it from the streamer and copy it into RAM, do your operations on it, then load in another slice.